File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/ShowFirstLayerBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.*
import com.usercentrics.sdk.flutter.serializer.deserializeBannerSettings
import com.usercentrics.sdk.flutter.serializer.serialize

internal class ShowFirstLayerBridge(
    private val assetsProvider: FlutterAssetsProvider,
    private val activityProvider: FlutterActivityProvider,
    private val bannerProxy: UsercentricsBannerProxy,
) : MethodBridge {

    override val name: String
        get() = "showFirstLayer"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)

        val argsMap = call.arguments as Map<*, *>

        bannerProxy.showFirstLayer(
            bannerSettings = argsMap["bannerSettings"].deserializeBannerSettings(
                assetsProvider,
                activityProvider,
            ),
        ) { response ->
            result.success(response?.serialize())
        }
    }
}

File: ios/usercentrics_sdk/Sources/usercentrics_sdk/Bridge/ShowFirstLayerBridge.swift
import UsercentricsUI
import Flutter

struct ShowFirstLayerBridge : MethodBridge {

    let assetProvider: FlutterAssetProvider
    let bannerProxy: UsercentricsBannerProxyProtocol
    let name: String = "showFirstLayer"

    init(assetProvider: FlutterAssetProvider,
         bannerProxy: UsercentricsBannerProxyProtocol = UsercentricsBannerProxy()) {
        self.assetProvider = assetProvider
        self.bannerProxy = bannerProxy
    }

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        let argsDict = call.arguments as! Dictionary<String, Any>
        let bannerSettings = BannerSettings(from: argsDict["bannerSettings"], assetProvider: assetProvider)
        bannerProxy.showFirstLayer(bannerSettings: bannerSettings) { response in
            result(response.serialize())
        }
    }
}
